python_modules/libraries/dagster-dg-cli/dagster_dg_cli/cli/scaffold/branch.py

```diff
@@ -136,27 +136,43 @@ def create_empty_commit(message: str) -> None:
     click.echo(f"Created empty commit: {message}")
 
 
-def create_content_commit_and_push(message: str) -> str:
+def create_content_commit_and_push(message: str, local_only: bool = False) -> str:
     _run_git_command(["add", "-A"])
     _run_git_command(["commit", "-m", message])
-    _run_git_command(["push"])
+    if not local_only and has_remote_origin():
+        _run_git_command(["push"])
     return _run_git_command(["rev-parse", "HEAD"]).stdout.strip()
 
 
-def push_branch_and_create_pr(branch_name: str, pr_title: str, pr_body: str) -> str:
+def has_remote_origin() -> bool:
+    """Check if the repository has a remote named 'origin'."""
+    result = subprocess.run(
+        ["git", "remote", "get-url", "origin"], capture_output=True, text=True, check=False
+    )
+    return result.returncode == 0 and bool(result.stdout.strip())
+
+
+def push_branch_and_create_pr(
```

Comment:
prob should rename this `create_branch_and_pr` since now a "push" doesn't always happen